src/hotspot/share/cds/lambdaFormInvokers.cpp

```diff
@@ -147,5 +170,36 @@ void LambdaFormInvokers::reload_class(char* name, ClassFileStream& st, TRAPS) {
 
   // exclude the existing class from dump
   SystemDictionaryShared::set_excluded(InstanceKlass::cast(klass));
+  SystemDictionaryShared::init_dumptime_info(result);
   log_info(cds, lambda)("Replaced class %s, old: %p  new: %p", name, klass, result);
 }
+
+void LambdaFormInvokers::dump_static_archive_invokers() {
+  if (_lambdaform_lines != nullptr && _lambdaform_lines->length() > 0) {
+    int count = _lambdaform_lines->length();
+    _static_archive_invokers = ArchiveBuilder::new_ro_array<Array<char>*>(count);
+    for (int i = 0; i < count; i++) {
+      char* str = _lambdaform_lines->at(i);
+      int len = (int)strlen(str);
+      Array<char>* line = ArchiveBuilder::new_ro_array<char>(len+1);
+      strcpy(line->adr_at(0), str); // copies terminating zero as well
```

Comment:
Should use strncpy for safety (code checkers will flag this).

Reply:
Will update with strncpy.